docs/cpp_docs.rst

```diff
@@ -128,77 +608,131 @@ For example:
 
-.. code-block:: cpp
+In addition, each value of the type should be documented. A short description is almost always sufficient.
 
-   /** Set a component's verbosity.
-   *
-   * If no verbosity is specified, inherit from parent
-   *
-   * @param name component of interest
-   * @param verbosity desired trace verbosity
-   */
-   void TraceImpl::setVerbosity(const std::string &name, const int verbosity) {
-   }
+For example:
 
-Annotating Arguments with Inline Comments (optional)
-----------------------------------------------------
+.. code-block:: cpp
 
-If the argument descriptions are very short, you may choose to annotate arguments with inline comments after each argument, one per line.
-These comments are prefixed with ``///< set:``.
+   /**
+    * Supported coordinate systems for flux-conserving transformations.
+    *
+    * These values are used in arguments to various functions in this package.
+    * Unless otherwise stated, these functions do not validate whether the data
+    * set makes sense in the "from" coordinates.
+    */
+   enum class CoordType
+   {
+       /** Untransformed detector coordinates. */
+       PIXEL,
+       /** Idealized detector coordinates after applying a distortion correction. */
+       WARP_PIXEL,
+       /** Equatorial J2000.0 coordinates. */
+       SKY_WCS
+   };
+
+.. _cpp-doxygen-method-function-structure:
+
+Documenting Methods and Functions
+==================================
 
-For example:
+All public or protected methods and all functions must be preceded by a documentation block.
+Method or function documentation blocks contain the following sections:
+
+1. :ref:`cpp-doxygen-short-summary`
+2. :ref:`cpp-doxygen-deprecation` (if applicable)
+3. :ref:`cpp-doxygen-extended-summary` (recommended)
+4. :ref:`cpp-doxygen-tparameters` (if applicable)
+5. :ref:`cpp-doxygen-parameters` (if applicable)
+6. :ref:`cpp-doxygen-returns` (if applicable)
+7. :ref:`cpp-doxygen-throws` (if applicable)
+8. :ref:`cpp-doxygen-exceptsafe`
+9. :ref:`cpp-doxygen-related` (if applicable; for functions only)
+10. :ref:`cpp-doxygen-see-also` (optional)
+11. :ref:`cpp-doxygen-notes` (optional)
+12. :ref:`cpp-doxygen-references` (optional)
+13. :ref:`cpp-doxygen-examples` (optional)
+
+An example of a Doxygen comment for a method:
 
 .. code-block:: cpp
 
    /** Set a component's verbosity.
-   *
-   * If no verbosity is specified, inherit from parent
-   */
-   void TraceImpl::setVerbosity(const std::string &name, ///< component of interest
-                                const int verbosity) { ///< desired trace verbosity
-   }
+    *
+    * If no verbosity is specified, it will be inherited from its parent.
+    *
+    * @param name component of interest
+    * @param verbosity desired trace verbosity
+    *
+    * @exceptsafe This method shall leave the component unchanged in the event of an exception.
+    */
+   void setVerbosity(std::string const & name, int verbosity);
```

Comment:
To make this more help as a quick reference, it might be a good idea to add (and document) a return value and some possible exceptions that could be thrown here.